nilearn/masking.py

```diff
@@ -126,6 +125,8 @@ def intersect_masks(mask_imgs, threshold=0.5, connected=True):
         grp_mask: 3D nibabel.Nifti1Image
             intersection of all masks.
     """
+    from .image import new_img_like
```

Comment:
Too many delayed imports is a sign of bad software design. First, you should not delay the import here but delay the import of masking in the image module, since it's used only one time. You should also check if there is a way to avoid these delayed imports by refactoring some code.